### PR TITLE
Snaxi Hotfixes

### PR DESCRIPTION
**Pull request by ghost**
Speedmerge thnx
should fix building in snow, 33.259103% tested. also minor other changes + apparently my ide likes adding tabs

:cl:
* rscadd: Snowmap changes. Fixed building tiles in the snow not deleting snowprints and blizzard fx. Lowered blizzard temps to not reach -70C. Adjusted blizzard durations to have a noticeable second stage. Fixed Cultify(), now it just stays snow. Nar-sie is chill. Changed changetile() to grab air from unsimulated tiles, fixing floors built on snow having no atmosphere for a moment. Fixed runtime errors.
* rscadd: Added asphalt (with blizzard fx) and noblizz_permafrost for our mappers to make stuff with.

### CODE REVIEW

**Comment by ghost** [comment]
Okay tested it actually and it works (apparently turfs change themselves to the new turf when built upon e.g. lattice instead of deleting and making a new one). Bandaid fix for runtimes doesnt realllllly work, I've figured out they only appear on mine tiles so I think plosky did something stupid.

**Comment by plosky1** [comment]
All the mine tiles are regular mine tiles so...yeah.

File: code/game/turfs/unsimulated/snow.dm [hunk 1]
```diff
@@ -21,11 +21,11 @@ var/blizzard_cooldown = 3000 //5 minutes minimum
 		command_alert(/datum/command_alert/blizzard_start)
 		sleep(rand(20 SECONDS, 2 MINUTES))
 		greaten_snowfall()
-		sleep(rand(50 SECONDS, 3 MINUTES))
+		sleep(rand(3 MINUTES, 6 MINUTES))
```
Comment (Kurfursten) [inline code/game/turfs/unsimulated/snow.dm:26]: Substance change not logged

File: code/game/turfs/unsimulated/snow.dm [hunk 2]
```diff
@@ -201,34 +210,34 @@ var/blizzard_cooldown = 3000 //5 minutes minimum
...
-			temperature = T_ARCTIC-15
+			temperature = T_ARCTIC-5
 			turf_speed_multiplier = 1.15 //For some reason, higher numbers mean slower.
 		if(SNOW_HARD)
-			temperature = T_ARCTIC-30
+			temperature = T_ARCTIC-10
 			turf_speed_multiplier = 1.6
 		if(SNOW_BLIZZARD)
-			temperature = T_ARCTIC-50
+			temperature = T_ARCTIC-20
```
Comment (Kurfursten) [inline code/game/turfs/unsimulated/snow.dm:222]: :cl: this too

File: code/game/turfs/unsimulated/snow.dm [hunk 3]
```diff
@@ -417,6 +425,10 @@ var/blizzard_cooldown = 3000 //5 minutes minimum
 	name = "permafrost"
 	desc = "Soil that never unfreezes."
 
+/turf/unsimulated/floor/snow/permafrost/New() //Band-aid fix for snow_state out of bounds exception
+	..()
+	snow_state = 0
```
Comment (Kurfursten) [inline code/game/turfs/unsimulated/snow.dm]: There has to be a better implementation. Will look into this later. Do you have the runtime?

**Review by Kurfursten — CHANGES_REQUESTED** [review]

File: code/game/turfs/unsimulated/snow.dm (hunk 3 above)
Comment (ghost) [reply code/game/turfs/unsimulated/snow.dm]: Yeah there's an out of bounds exception in /proc/cache_snowtile()
for(var/i = 1 to overlay_counts[parent.snow_state+1])
overlay_counts is out of bounds. I checked and apparently the snow_state gets up to 6, if you look at greaten_snowfall and lessen_snowfall I'm not really sure how. The New() fix didn't seem to work.

Comment (ghost) [reply code/game/turfs/unsimulated/snow.dm]: Also, this only happens on mine tiles. Not the permafrost tiles that make up the pathways

Comment (Kurfursten) [reply code/game/turfs/unsimulated/snow.dm]: Just paste the whole runtime here or in a hastebin

**Comment by SonixApache** [comment]
did you cut the 5 minute sound files into smaller bites yet so it doesn't lag every time someone steps outside?

**Comment by ghost** [comment]
It doesn't lag after it gets loaded. If there isn't a way to load it normally, I'm just going to make a proc that instantly plays a sound then stops called load_sound(), then set snowstorms to load on roundstart

**Comment by SonixApache** [comment]
or you could just cut the sound file into a bunch of more manageable files

**Comment by ghost** [comment]
Why don't YOU cut admin ogg's into managable files, those are 5 mins long.
It's honestly strange that it lags at all, thanks byond

**Comment by ghost** [comment]
Alright, I fixed everything except the minor lag whenever hearing snowstorm sound for the first time.

**Comment by SonixApache** [comment]
what you could do is cut the 5 minute snowstorm sound file into a bunch of managable bits that have fade-ins and fade-outs - that way you could play them randomly to prevent sameysound AND not lag people at the same time

**Comment by ghost** [comment]
In another PR, sure. But this has important HOTfixes in it.

File: code/game/turfs/turf.dm [hunk 4]
```diff
@@ -353,14 +353,16 @@
 		for(var/obj/effect/decal/cleanable/C in src)
 			qdel(C)//enough with footprints floating in space
 
-	if(istype(src,/turf/simulated))
+	if(istype(src,/turf))
```
Comment (MadmanMartian) [inline code/game/turfs/turf.dm]: This will never not be true

Comment (ghost) [reply code/game/turfs/turf.dm]: True.

**Pull request merged**